src/applications/personalization/profile/tests/e2e/personal-and-contact-info/modals.cypress.spec.js

```diff
@@ -143,4 +144,37 @@ describe('Modals on the personal information and content page', () => {
       sectionName: 'contact email address',
     });
   });
+
+  it('should clear transactions when successful transaction occurs', () => {
```

Comment:
I love everything about this PR but would suggest a couple very small tweaks:

1. Move this new test into its own `describe` block. I think it's fine for the test to live in this file, but this particular test isn't really testing the warning modals like all of the other tests.
2. And make the test description really clear that it's testing the ability to re-edit a piece of contact info after making a no-op save